end-to-end-tests/echo_test.go

```diff
@@ -0,0 +1,205 @@
+	time.Sleep(2 * time.Second)
+
+	// Fix the Blochain timestamp. Must be "in the future"
+	err = SetNextBlockTimestamp(ctx, LocalBlockchainHttpEndpoint, 7000000000)
+	require.NoError(s.T(), err)
+
+	// Run Node Service
+	nodeConfig := NewLocalNodeConfig()
+
+	nodeConfig.SnapshotDir = tempDir
+
+	ctx, cancel := context.WithCancel(ctx)
+	supervisor, err := node.Setup(ctx, nodeConfig)
+	require.NoError(s.T(), err)
+
+	ready := make(chan struct{}, 1)
+	supervisorErr := make(chan error, 1)
+	go func() {
+		err := supervisor.Start(ctx, ready)
+		if err != nil {
+			supervisorErr <- err
+		}
+	}()
+
+	select {
+	case err := <-supervisorErr:
+		require.NoError(s.T(), err)
+	case <-ready:
+		break
+	}
+
+	// Configure Suite for tear down
+	s.containers = depsContainers
+	s.tempDir = tempDir
+	s.cancel = cancel
+	s.supervisorErr = supervisorErr
+
+}
+
+func (s *EchoInputTestSuite) TearDownSuite() {
+
+	// Stop Node services
+	s.cancel()
+
+	// Remove machine snpshot
+	os.RemoveAll(s.tempDir)
+
+	// Terminate deps
+	ctx := context.Background()
+	err := deps.Terminate(ctx, s.containers)
+	require.NoError(s.T(), err)
+
+	// Clear default_tx_database
+	cwd, err := os.Getwd()
+	require.NoError(s.T(), err)
+
+	_ = os.Remove(filepath.Join(cwd, defaulTxDatabaseFile))
+
+}
+
+func (s *EchoInputTestSuite) SetupTest() {
+
+}
+
+func (s *EchoInputTestSuite) TestSendInput() {
+
+	ctx := context.Background()
+
+	inputIndex, err := AddInput(ctx, LocalBlockchainHttpEndpoint, payload)
+	require.NoError(s.T(), err)
+
+	// Check input was correctly added to the blockchain
+	require.EqualValues(s.T(), 0, inputIndex)
+
+	require.NoError(s.T(), AdvanceTime(ctx, LocalBlockchainHttpEndpoint, 120))
+
+	// Get Input with vouchers and proofs
+	graphQlClient := graphql.NewClient("http://localhost:10000/graphql", nil)
+	getInputChan := make(chan *readerclient.Input, 1)
+	getInputErr := make(chan struct{}, 1)
+
+	go func() {
+		var resp *readerclient.Input
+		attempts := 0
+		for ; attempts < maxReadInputAttempts; attempts++ {
+			time.Sleep(2 * time.Second)
+			resp, err = readerclient.GetInput(ctx, graphQlClient, inputIndex)
+			if err == nil && resp.Status == "ACCEPTED" &&
+				resp.Vouchers != nil && resp.Vouchers[0].Proof != nil &&
+				resp.Notices != nil && resp.Notices[0].Proof != nil {
+				break
+			}
+		}
+		if attempts == maxReadInputAttempts {
+			getInputErr <- struct{}{}
+			return
+		}
+		getInputChan <- resp
+	}()
+
+	select {
+	case input := <-getInputChan:
+
+		//Check Input
+		expectedInput, err := readExpectedInputWithProof("expected_input_with_proofs.json")
+		require.NoError(s.T(), err)
+
+		require.EqualValues(s.T(), expectedInput, input)
+
+		break
+	case err = <-s.supervisorErr:
+		require.NoError(s.T(), err)
+		break
+	case <-getInputErr:
+		require.FailNow(s.T(), "Could not retrieve Voucher with Proof 0")
+	}
+
+}
+
+func readExpectedInputWithProof(filename string) (*readerclient.Input, error) {
+	cwd, err := os.Getwd()
```

Comment:
You can use the embed package to add these files to the node binary. See https://pkg.go.dev/embed